make default image become a template var

jianlinliu opened the pull request:
When trying to deploy the saml app on AWS/GCE, due to https://bugzilla.redhat.com/show_bug.cgi?id=1316341#c6, QE have to specify saml image using registry.qe.openshift.com/openshift3/saml-service-provider instead of openshift3/saml-service-provider. This PR would make saml app deployment more flexible.

@brenton could you help have a review.

jianlinliu committed:
make default image become a template var

brenton commented:
This looks great Jianlin.  With this change are you able to pull from an authenticated registry if the Kubelet's authentication is set up correctly?

jianlinliu commented:
Yes, I already tested it, it works.

jianlinliu committed:
remove stale health check

For 3.2, when deploying saml-auth pod, get the following warning, but pod could become "running" status.
# oc describe po saml-auth-1-dx3qo
...
  57m		57m		3	{kubelet xxx.example.com}	spec.containers{saml-auth}	Warning		Unhealthy	Readiness probe failed: Get https://10.1.0.6:8443/logged_out.html: dial tcp 10.1.0.6:8443: connection refused

While for 3.3, the failed health check make the pod ended with "error".
# oc get po
NAME                      READY     STATUS    RESTARTS   AGE
saml-auth-2-d70n7         0/1       Running   0          4m
saml-auth-2-deploy        1/1       Running   0          5m

# oc describe po saml-auth-2-d70n7
  5s		5s		1	{kubelet xxx.example.com}	spec.containers{saml-auth}	Warning		Unhealthy	Readiness probe failed: Get https://10.1.1.3:8443/logged_out.html: local error: no renegotiation

# oc get po
NAME                      READY     STATUS    RESTARTS   AGE
saml-auth-2-deploy        0/1       Error     0          13m

Seem like the health check is stale, so suggest remove it, or correct it.

briantward commented:
@brenton This can be closed as it was made obsolete by PR #26
